### ISSUE TITLE
support viral searches with gather by allowing other thresholding?

### ISSUE THREAD

**Issue by ctb** [sourmash-bio/sourmash#1841]
over in genome-grist, https://github.com/dib-lab/genome-grist/pull/130 made it possible to search private databases, and so we've been trying out phage database searches.

a problem that has cropped up with sourmash as a result is that `sourmash gather` uses `--threshold-bp` to truncate gather searches at some point. This is less than good for short sequences in the reference database - for example, with a threshold bp of 10,000, a 10kb virus with a 95% match will not be output, while a 20kb virus with a 55% match will. I can see situations where we would like the latter to be allowed!

this could be done with alternate thresholding. how would that work?

looking at search.py, class `GatherDatabases` uses `self.threshold_bp` to truncate searches in `find_best(...)`. This is because gather counts things in terms of estimated overlapping bp, so it's a natural kind of thresholding. This was probably more important in the days before prefetch, when gather was pretty slow... but now most of the work is done before we even get to the gather class, so it's probably less important (but see https://github.com/sourmash-bio/sourmash/issues/1771, which suggests some performance issues that need to be looked at).

Given the way the greedy min-set-cov algorithm works in gather, which is based on overlapping bp, it seems like there is no performance benefit to trying to filter searches within the gather algorithm itself.

So, basically, the only way to enable this would be to support post-gather filtering - use a low `--threshold-bp`, and choose your scaled wisely, then get all the matches.

I'm not sure if that should be done within sourmash as an alternate command-line flag - either something based on (estimated) ANI, or estimated coverage/containment of the matching genome? - or be done within genome-grist. Maybe we can try over in genome-grist and if it's useful bring it into sourmash? Although the ANI threshold might be useful / resonate with users.

Calling in @bluegenes because I used the word "ANI" - any thoughts?

**Comment by ctb** [comment]
conversation with @hehouts and @bluegenes suggests that they do not strongly disagree with the opinions expressed above 😄 . leaving this for posterity!
